### ISSUE TITLE
Ideas on how to wrap Shmex.Native reads to a process?

### ISSUE THREAD

**Issue by eliasdarruda** [membraneframework/membrane_core#767]
I wanted to link %Shmex{} to a process and receive writes in my message queue using receive or handle_info.

Is this a valid use-case for Shmex?

**Comment by mat-hek** [comment]
👋 Do you mean you want to get a message when someone writes to the shared memory? Shmex doesn't support this. I'm unsure if this is even possible just with shared memory - you'd rather need a separate communication channel for that purpose.

**Comment by eliasdarruda** [comment]
> 👋 Do you mean you want to get a message when someone writes to the shared memory? Shmex doesn't support this. I'm unsure if this is even possible just with shared memory - but you can do it with a separate communication channel.

Exactly, do you have any examples of a separate communication channel reading through the shared memory?

First thing that I thought of was to have a callback that the NIF will call whenever some kind of synchronization byte changes based on a loop looking at the shared memory. But that would need to be implemented in the NIF, I'm thinking if I can use it as is but still having the possibility of getting this message somehow

**Comment by mat-hek** [comment]
> Exactly, do you have any examples of a separate communication channel reading through the shared memory?

I meant that whoever else is writing to the shared memory can send a message via a socket / port / erlang distribution if it's an erlang node whenever they do the write. And when you have a socket / port / erlang distribution make sure that it's still worth having shared memory too :D

> First thing that I thought of was to have a callback that the NIF will call whenever some kind of synchronization byte changes based on a loop looking at the shared memory

I'm not sure if it's the optimal approach, but I guess that by using Shmex you can write this loop in Elixir

**Comment by eliasdarruda** [comment]
Yeah I guess doing that would kill the need for shared memory xD

My use case is simple, I need a way to communicate between two services that are not written in the same language, but I need to be able to listen to messages to react to it later. I want to avoid TCP and also tried with Ports and led to some inconsistencies with IO, thats when I wen't looking for a shared memory approach.

Thank you for the quick reply!

**Comment by mat-hek** [comment]
I guess that both ports and TCP would be easier to handle and not much slower than shared memory, so unless performance is the top priority I would consider them again ;)
